src/scripts/Controller.js

```diff
@@ -719,13 +730,57 @@ export default class SitemapController {
 		await this.store.logOut().finally(async () => await this.showAuthPage());
 	}
 
-	async showSitemaps() {
+	async showProjects() {
+		this.clearState();
+		this.setActiveNavigationButton('projects');
+
+		$(`#sitemaps-nav-button`).prop('disabled', true);
+		$('#create-sitemap-nav-button').prop('disabled', true);
+
+		const projects = await this.store.getAllProjects();
+
+		if (projects.error_msg) {
+			$('#sitemaps').hide();
+			$('#viewport').html(
+				'<div class="container"><div data-i18n="get_sitemap_error"></div></div>'
+			);
+			Translator.translatePage();
+		}
+
+		const $projectListPanel = ich.ProjectList();
+
+		$('.nav').show();
+		if (this.store.supportAuth) {
+			$('#auth_nav').css('display', 'block');
+		} else {
+			$('#auth_nav').css('display', 'none');
+		}
+		projects.forEach(project => {
+			const $project = ich.ProjectListItem(project);
+			$project.data('project', project);
+			$projectListPanel.find('tbody').append($project);
+		});
+		$('#viewport').html($projectListPanel);
+		Translator.translatePage();
+	}
+
+	async showSitemaps(projectId, projectName) {
 		this.clearState();
-		this.setActiveNavigationButton('sitemaps');
 
-		const sitemaps = await this.store.getAllSitemaps();
+		if (typeof projectId === 'string') {
+			$('#sitemaps-nav-button span#navbar-active-project-id').text(`(${projectName})`);
+			$(`#sitemaps-nav-button`).attr('projectid', projectId);
+		} else {
+			projectId = $(`#sitemaps-nav-button`).attr('projectid');
```

Comment:
Move project id extraction into method similar to getCurrentlyEditedXXX, or populate controller state with it.

Reply:
done